indexer/ai/src/main/java/com/google/cloudsearch/ai/StandardSkillCategoryExtraction.java

```diff
@@ -0,0 +1,180 @@
+        }
+    }
+
+    /**
+     * Set the filters required for Category Extraction.
+     * @param filter    JSON Object Containing the filters required for Category Extraction.
+     */
+    @Override
+    public void setFilter(JSONObject filter) {
+        this.categoryConfidence = (double) filter.get(Constants.CONFIG_CATEGORY_CONFIDENCE);
+    }
+
+    /**
+     * Returns the filters
+     * @return  Returns the filters in JSON Format.
+     */
+    @Override
+    public JSONObject getFilter() {
+        JSONObject obj = new JSONObject();
+        obj.put(Constants.CONFIG_CATEGORY_CONFIDENCE, this.categoryConfidence);
+        return obj;
+    }
+
+  /**
+   * Parse the JSON Object containing the filters for the skill and validate it before calling the setter.
+   * @param filter  JSON Object containing the filters
+   * @throws InvalidConfigException     Throws exception if the filter name is not supported.
+   */
+  @Override
+  public void parseFilter(JSONObject filter) throws InvalidConfigException {
+        if(filter == null) {
+            return;
+        }
+        for(Object key : filter.keySet()) {
+            if (key.equals(Constants.CONFIG_CATEGORY_CONFIDENCE))
+                setFilter(filter);
+            else {
+                throw new InvalidConfigException("Filter " + key + " is not supported for AISkill Category Extraction.");
+            }
+        }
+    }
+
+    /**
+     * Check if the confidence score satisfied the specified filter.
+     * @param confidence    Confidence score obtained for each category returned by the API
+     * @return              Returns true if the condition is satisfied, else returns false.
+     */
+    private boolean isFilterSatisfied(double confidence) {
+        if(confidence >= this.categoryConfidence)
+            return true;
+        else
+            return false;
+    }
+
+    /**
+     * Parses the Configuration for the Category Extraction Skill
+     * @param aiSkill        JSONObject for the skill
+     * @param schema         CloudSearch schema
+     * @throws InvalidConfigException
+     */
+    @Override
+    protected void parse(JSONObject aiSkill, JSONObject schema) throws InvalidConfigException {
+        parseAISkillName((String) aiSkill.get(Constants.CONFIG_SKILL_NAME));
+        parseOutputMappings((JSONArray) aiSkill.get(Constants.CONFIG_OUTPUT_MAPPINGS), schema);
+        parseInputs((JSONObject) aiSkill.get(Constants.CONFIG_INPUTS));
+        parseFilter((JSONObject) aiSkill.get(Constants.CONFIG_FILTERS));
+    }
+
+    /**
+     * Execute the Category Extraction Skill and populates the structured Data.
+     * @param contentOrURI      The actual file content or Cloud storage URI
+     * @param structuredData    Multimap to store the structured data for indexing.
+     */
+    @Override
+    public void executeSkill(String contentOrURI, Multimap<String, Object> structuredData) {
+        try (LanguageServiceClient language = LanguageServiceClient.create()) {
```

Comment:
nit: lets call it languageService

Reply:
Also this client should be created only once ?

Reply:
I have changed the name, if we want to create it once it should be created somewhere in the skill driver before executing and needs to be passed to execute method, if that is fine I can change this.